### PR TITLE
[Optimism] Feature/custom network gas estimates

### LINKED ISSUE #2864: [Optimism] Custom Network Gas Support
- Need to provide custom networks with a gas estimate directly from the blockchain.

- Involves updates to the Transaction Controller and gasEstimateMethods in the the app

- Test methods can be found in Slack channel

  - Review gas estimates for both _mainnet_ and _custom networks_
  - In all txs (send, deep link & DAPP tx)

### PR DESCRIPTION
**Pull request by sethkfman**
**Description**

Leverage blockchain gas estimation fees for custom networks.

**Checklist**

* [X] There is a related GitHub issue
* [X] Tests are included if applicable
* [X] Any added code is fully documented

**Issue**

Resolves #2864

### CODE REVIEW

**Comment by sethkfman** [comment]
Build is currently failing due to local dependency of the controller tgz. Once, I complete the controllers PR this should be passing.

File: app/core/Engine.js [hunk 1]
```diff
@@ -139,7 +137,15 @@ class Engine {
...
-					getAssetsState: () => assetsController.state
+					getAssetsState: () => assetsController.state,
+					getTokenListState: () => {
+						if (contractMetadata) {
+							return {
+								tokenList: Object.values(contractMetadata).filter(token => token.erc20)
+							};
+						}
+						return { tokenList: [] };
+					}
```
Comment (wachunei) [inline app/core/Engine.js:148]: This is what we should adopt to land v11 on the mobile app in the meantime right?

Comment (sethkfman) [reply app/core/Engine.js:148]: Yeah, it was a suggestion from @Gudahtt. It provides us with something while we wait for the more robust list.

Comment (wachunei) [reply app/core/Engine.js:148]: We can refactor to this and add a `// TODO:` to mark we need to change it in the future
```ts
getTokenListState: () => ({ tokenList: Object.values(contractMetadata || []).filter(token => token.erc20) }),
```

Comment (Gudahtt) [reply app/core/Engine.js:148]: Oh, I just noticed one issue - that `tokenList` should be an object. See the `tokenMap` type: https://github.com/MetaMask/controllers/blob/main/src/assets/TokenListController.ts#L36

Something like 

```javascript
const tokens = {}
for (const address in contractMetadata)  {
  if (contractMetadata[address].erc20) {
    tokens[address] = contractMetadata[address]
  }
}
```

Or with lodash:

```
pickBy(contractMetadata, (metadata) => metadata.erc20)
```

Comment (sethkfman) [reply app/core/Engine.js]: I made a few tweaks and added the TODO

**Comment by wachunei** [comment]
Just to keep in mind for the future: I expect we start relying on `GasFeeController` which has `basicGasEstimates` for custom networks, BSC initially (once metaswaps API is deployed), so checking for `isCustomNetwork` in the future should change to checking wether is a network we support with `GasFeeController` (preferably the same check we do with `getCurrentNetworkLegacyGasAPICompatibility` in `GasFeeController` constructor)

**Review by wachunei — APPROVED** [review]
LGTM

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍🏽

File: yarn.lock [hunk 2]
```diff
@@ -11409,9 +11068,9 @@ react-native-view-shot@^3.1.2:
   integrity sha512-9u9fPtp6a52UMoZ/UCPrCjKZk8tnkI9To0Eh6yYnLKFEGkRZ7Chm6DqwDJbYJHeZrheCCopaD5oEOnRqhF4L2Q==
 
 react-native-webview@^11.0.2:
-  version "11.0.2"
-  resolved "https://registry.yarnpkg.com/react-native-webview/-/react-native-webview-11.0.2.tgz#c88e84fa470f04ff070ff8a2c7f4d7295d46bb06"
-  integrity sha512-GDyIBRbCZ2wbMUGCxA7LufSEbSoWKOzkFB8YljmAffA15tzN6ccvGEquB/hkk5KhvoYy300kwJyEmyBeG6d/AA==
+  version "11.6.5"
```
Comment (Gudahtt) [inline yarn.lock:11071]: The `react-native-webview` patch is out-of-date as of this change, resulting in an error during installation

**Pull request merged**